### DISCUSSION TITLE
Unstable behaviour with cc2531 -> crash?!

### DISCUSSION THREAD

**Discussion by Rico81** [Koenkk/zigbee2mqtt#6430] [Devices]
Hello 

I've been using an rpi 4 with cc2531 for a couple of weeks now and encountered some problems concerning the stability.
The system worked well but regularily stopped after a week (after the last reboot) or so. Altough that was really annoying I could at least get it back with rebooting.
Yesterday the system hang and would not come back with rebooting.
On the node red ui the system is responding, shows consumptions, sends signals and so on. And it seems as if wifi devices continue to work, at least a belkin socket does so with a time schedule. But all the zigbee devices are not working anymore. 
In the zigbee2mqtt service there are timeout errors (from the commands i've tried after rebooting) and a line "cannot request when znp has not initialized yet". 
So what has happend here? Is the firmware of the cc2531 broken?

Grateful for any hint.

Cheers
Rico

**Comment by Koenkk** [comment]
You probably need to update to a more powerful adapter: https://www.zigbee2mqtt.io/getting_started/what_do_i_need.html#supported-zigbee-adapter

**Reply by Rico81** [reply]
Thanks for your answer. I will happily upgrade my setup with one of these sticks if that is the solution.
But only for my own understanding: How can to little power cause an adapter to suddenly become unable 
to connect with certain devices but continue to work with others?
Or is it that these cc2531 adapters have just proven to be buggy?

**Reply by Koenkk** [reply]
CC2531 cannot handle many devices, before switching an adapter you can also try the following firmware which provides more stability: https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_Home_1.2/bin/source_routing
